File: heart/src/main/java/com/vantagecircle/heartrate/activity/BaseActivity.java
package com.vantagecircle.heartrate.activity;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

/**
 * Created by bapidas on 09/10/17.
 */

public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setActivityComponent();
    }

    protected abstract void setActivityComponent();

    protected abstract void init();
}

File: app/src/main/java/com/bapidas/heartrate/ui/base/BaseActivity.kt
package com.bapidas.heartrate.ui.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

/**
 * Created by bapidas on 09/10/17.
 */
abstract class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setActivityComponent()
    }

    protected abstract fun setActivityComponent()
    protected abstract fun init()
}